packages/hierarchy-builder/src/test/HierarchyProvider.test.ts

```diff
@@ -849,6 +849,7 @@ describe("HierarchyProvider", () => {
                 fullClassName: "x.y",
                 query: { ecsql: "QUERY" },
               },
+              { node: { key: "root", label: "root" } },
```

Comment:
why was this necessary?

Reply:
It wasn't necessary for this test specifically, but it now does exactly what needs to be done to get coverage for the `else` part [https://github.com/iTwin/presentation/blob/cbdf76ad832214d2fb2ef59d713c257bc3600668/packages/hierarchy-builder/src/hierarchy-builder/HierarchyProvider.ts#L436C7-L436C47](url).

Reply:
> It wasn't necessary for this test specifically

That's exactly what's wrong with it. The change makes the line covered, but doesn't check if the covered line actually did anything or affected the result in any way, which it should. 

Options in the order of preference:
1. Write a test that tests the situation specifically
2. Restructure the library code to not require a specific test for the line to be covered
3. `// istanbul ignore next`

Unless you have an idea for a test, I think here the second option is fine - doing a `return lhs.length - rhs.length` should do the trick.

Reply:
> That line of code is needed for the Dictionary to work correctly, and I don't think that we should be testing something that we use from other packages.

I agree with both statements, but that piece of code is ours, not of some other package. Which means it should be tested for correctness on our end. Having said that, sometimes having a test for a very trivial piece of code is just not worth it.